Framework/DataHandling/src/LoadPSIMuonBin.cpp

```diff
@@ -356,185 +470,321 @@ void LoadPSIMuonBin::assignOutputWorkspaceParticulars(
+  } else if (std::find(std::begin(psiMonths), std::end(psiMonths),
+                       line.substr(5, 3)) != std::end(psiMonths)) {
+    // If the line contains a Month in the PSI format then assume it conains a
+    // date on the line. 5 is the index of the line that is where the month is
+    // found and 3 is the length of the month.
+    processDateHeaderLine(line);
+  }
+}
+
+void LoadPSIMuonBin::readInTemperatureFileHeader(const std::string &contents) {
+  const int uselessLines = 6;
+  int lineNo = 0;
+  std::string line = "";
+  for (const auto charecter : contents) {
+    if (charecter == '\n') {
+      if (line[0] == '!' && lineNo > uselessLines) {
+        processHeaderLine(line);
+      } else if (line[0] != '!') {
+        return;
+      }
+      ++lineNo;
+      line = "";
+    } else {
+      line += charecter;
+    }
+  }
+}
+
+void LoadPSIMuonBin::processLine(const std::string &line,
+                                 DataObjects::Workspace2D_sptr &ws) {
+  std::vector<std::string> segments;
+  boost::split(segments, line, boost::is_any_of("\\"));
+
+  const auto recordTime = segments[0];
+  const auto numValues = std::stoi(segments[1]);
+  std::vector<std::string> firstValues;
+  boost::split(firstValues, segments[2], boost::is_any_of(" "));
+  std::vector<std::string> secondValues;
+  boost::split(secondValues, segments[3], boost::is_any_of(" "));
+
+  // example recordTime = 10:10:10
+  // 10 hours, 10 minutes, and 10 seconds. Hence the substr choices here.
+  double secondsInRecordTime =
+      (std::stoi(recordTime.substr(0, 2)) * 60 * 60) + // Hours
+      (std::stoi(recordTime.substr(3, 2)) * 60) +      // Minutes
+      (std::stoi(recordTime.substr(6, 2)));            // Seconds
+  const auto timeLog = (Types::Core::DateAndTime(m_tempHeader.startDateTime) +
+                        secondsInRecordTime)
+                           .toISO8601String();
+
+  Mantid::API::Algorithm_sptr logAlg = createChildAlgorithm("AddTimeSeriesLog");
+  logAlg->setProperty("Workspace", ws);
+  logAlg->setProperty("Time", timeLog);
+  if (!m_tempHeader.delimeterOfTitlesIsBackSlash) {
+    for (auto i = 0; i < numValues; ++i) {
+      logAlg->setProperty("Name", "Temp_" + m_tempHeader.titles[i]);
+      logAlg->setProperty("Type", "double");
+      logAlg->setProperty("Value", firstValues[i]);
       logAlg->executeAsChildAlg();
     }
+  } else {
+    logAlg->setProperty("Name", "Temp_" + m_tempHeader.titles[0]);
+    logAlg->setProperty("Type", "double");
+    logAlg->setProperty("Value", firstValues[0]);
+    logAlg->executeAsChildAlg();
+
+    logAlg->setProperty("Name", "Temp_" + m_tempHeader.titles[1]);
+    logAlg->setProperty("Type", "double");
+    logAlg->setProperty("Value", secondValues[0]);
+    logAlg->executeAsChildAlg();
+  }
+}
+
+std::string LoadPSIMuonBin::detectTempFile() {
+  const std::string binFileName = getPropertyValue("Filename");
+  const Poco::Path fileDir(Poco::Path(binFileName).parent());
```

Comment:
Replace Poco with boost::filesystem here

Reply:
did you make this change?